src/Frame.h

```diff
@@ -0,0 +1,35 @@
+#ifndef __FRAME_H__
+#define __FRAME_H__
+
+#include <string>
+#include <cstdint>
+
+using namespace std;
+
+class Frame {
+public:
+    // Getters
+    uint8_t src();
+    uint8_t dst();
+    uint8_t size();
+    string data();
+
+    // Setters
+    void set_src(uint8_t src);
+    void set_dst(uint8_t dst);
+    void set_data(string data);
+    void append(string extradata);
+    void clear();
+
+    // Constructors
+    Frame();
+    Frame(string contents);
+    Frame(uint8_t src, uint8_t dst, string data);
```

Comment:
Or alternatively, explicitly delete them if we decide we don't want this object to either be copyable, moveable, or both.